fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/query/FDBReturnedRecordLimitQueryTest.java

```diff
@@ -173,17 +171,9 @@ void testComplexLimits6() throws Exception {
 
         // Scan(<,>) | [MySimpleRecord]
         RecordQueryPlan plan = planner.plan(query);
-        if (planner instanceof RecordQueryPlanner) {
-            assertThat(plan, typeFilter(contains("MySimpleRecord"), scan(unbounded())));
-            assertEquals(1623132336, plan.planHash(PlanHashable.PlanHashKind.LEGACY));
-            assertEquals(-1229687959, plan.planHash(PlanHashable.PlanHashKind.FOR_CONTINUATION));
-            assertEquals(-1229687959, plan.planHash(PlanHashable.PlanHashKind.STRUCTURAL_WITHOUT_LITERALS));
-        } else {
-            assertThat(plan, typeFilter(contains("MySimpleRecord"), scan(unbounded())));
-            assertEquals(1623132336, plan.planHash(PlanHashable.PlanHashKind.LEGACY));
-            assertEquals(667716470, plan.planHash(PlanHashable.PlanHashKind.FOR_CONTINUATION));
-            assertEquals(667716470, plan.planHash(PlanHashable.PlanHashKind.STRUCTURAL_WITHOUT_LITERALS));
-        }
+        assertThat(plan, typeFilter(contains("MySimpleRecord"), scan(unbounded())));
+        assertEquals(1623132336, plan.planHash(PlanHashable.CURRENT_LEGACY));
+        assertEquals(-145642685, plan.planHash(PlanHashable.CURRENT_FOR_CONTINUATION));
```

Comment:
This plan hash also appears to have changed

Reply:
collapsed `if`

Reply:
New: `-145642685`
Old: `-1229687959` (with `ReocrdQueryPlanner`) or `667716470` (with Cascades)